src/vos/vos_common.c

```diff
@@ -346,6 +378,18 @@ vos_init(void)
 		D_GOTO(out, rc);
 	}
 
+	/* Some cases (DAOS-4310) have been found where the default ABT's ULTs
+	 * stack-size (16K) is not enough and this can lead to stack overrun
+	 * and corruption, so double it just in case
+	 */
+	if (abt_thread_stacksize() < 32768) {
+		rc = set_abt_thread_stacksize(32768);
+		if (rc != 0) {
+			D_ERROR("failed to set ABT_THREAD_STACKSIZE: %d\n", rc);
+			return rc;
```

Comment:
Forgot to unlock mutex?

Reply:
Oops, good catch ! Seems that I have replicated same code from abc_init() a bit too quickly... Will fix in next commit